cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -220,16 +229,27 @@ public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
         @Override
         public AnnotationVisitor visitParameterAnnotation(int parameter, String desc, boolean visible) {
           if (!(authEnforceAnnotationNode != null && visible &&
-            Name.class.getName().equals(Type.getType(desc).getClassName()))) {
+                  (SUPPORTED_PARAMETER_ANNOTATIONS.contains(Type.getType(desc).getClassName())))) {
             return null;
           }
           // Since AuthEnforce annotation was used on this method then look for parameters with named annotation and
           // store their position.
-          // TODO: Should also pick up names from QueryParam and PathParam annotations here as needed later and also
           // decide on a precedence order.
           AnnotationNode annotationNode = new AnnotationNode(desc);
-          // store the parameter position and its annotation detail
-          parameterAnnotationNode.put(parameter, annotationNode);
+          // store the parameter position and its type with annotation detail
+          ParameterDetail parameterDetail = new ParameterDetail(parameterTypes[parameter], annotationNode);
+          if (parameterDetails.containsKey(parameter)) {
+            // we already stored some annotation for this parameter update it if the current one is Name annotation
+            // as its preferred
+            if (Name.class.getName().equals(Type.getType(desc).getClassName())) {
+              LOG.info("Updating parameter details for parameter at position '%s' for method '%s' in class '%s' from " +
```

Comment:
Logger should be using `{}` as placeholder, not `%s`. Also, this should be at `DEBUG` level.